source/material_model/visco_plastic.cc

```diff
@@ -630,6 +645,12 @@ namespace aspect
                                                                           n_fields,
                                                                           "Thermal diffusivities");
 
+          define_conductivities = prm.get_bool ("Define thermal conductivities");
+
```

Comment:
Here add the following:
1. An Assert that makes sure at most one of thermal diffusivities or thermal conductivities was changed from the default value.
2. If both are at default values or if only thermal conductivities are set, execute the line below.
3. If only thermal diffusivities are set differently from the default value, use line 644 to read thermal diffusivities in a temporary vector, then convert and store them using density and cp into `thermal conductivities`.

This should take care of every case existing parameter files and new files, without introducing a temporary input parameter that we would need to remove later.

Reply:
A bit confused here - one of the comments requests `thermal_diffusivities` be removed from `visco_plastic.h` ? Was that correct or did you mean remove `thermal_conductivites`?

Maybe the simplest thing is to just switch to solely using thermal conductivity now as that is what we will do in the long run regardless?

Reply:
Yes, I think we agreed to remove `thermal_diffusivities` as a member variable from visco_plastic.h, but keep the input parameter for now (and store it in a temporary vector inside of `parse_parameters` to then convert it to thermal conductivities and store it in `thermal_conductivities`). This way we keep the code simpler and maintain parameter file compatibility for now.

Reply:
Thanks and glad you had a proper recollection of the discussion. This will likely change the test results due to the order of averaging thermodynamic properties, but I'm fine with that if you are as well.

I'm still a bit confused about the proposed Assert in your step 1 above - it seems like this requires parsing the thermal conductivities first? Or would this be an Assert that checks whether the parameter is simply declared in the parameter file?

Reply:
I would write it like this, which would check that at least one of the two parameters is set to the default value:
```
AssertThrow (prm.get("Thermal conductivities") == "3.0" || prm.get("Thermal diffusivities") == "0.8e-6", ExcMessage("Both thermal conductivities and thermal diffusivities have been set. This is not possible. Please only specify thermal conductivities as the input parameter 'thermal diffusivities' will be removed in a future version."));
```